src/classes/CW_CampaignTriggerHandlerTest.cls

```diff
@@ -0,0 +1,266 @@
+		system.assertEquals(10.0, roadmaps[0].Points_Used__c);
+		
+		// update Campaign to remove lookup to Roadmap
+		camps[0].Roadmap__c = null;
+		update camps[0];
+		
+		// assert campaign (child which does not have a Roadmap) points are not rolled up to roadmap (parent)
+		roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(5.0, roadmaps[0].Points_Used__c);
+		
+		// update Campaign to add lookup to Roadmap
+		camps[0].Roadmap__c = roadmap.Id;
+		update camps[0];
+		
+		// assert campaign (child) points are rolled up to roadmap (parent)
+		roadmaps = [select Id, Points_Used__c from CW_Roadmap__c where Id = :roadmap.Id];
+		system.assertEquals(10.0, roadmaps[0].Points_Used__c);
+	}
+	
+	@isTest static void updateChildSchedulesWithCampaignStageTest() {
+		
+		//setup data
+		List<CW_Campaign__c> camps = TestDataUtil.getCampaigns(2);
+		for (CW_Campaign__c c :camps) c.Stage__c = 'stage 1';
+		
+		// insert Campaign (with stage)
+		insert camps;
+		
+		List<CW_Schedule__c> schedules = new List<CW_Schedule__c>();
+		List<CW_Schedule__c> schedules1 = TestDataUtil.getSchedules(camps[0].Id, 5);
+		List<CW_Schedule__c> schedules2 = TestDataUtil.getSchedules(camps[1].Id, 5);
+		schedules.addAll(schedules1);
+		schedules.addAll(schedules2);
+		
+		// insert schedules
+		insert schedules;
+		
+		// get schedule ids
+		Set<Id> scheduleIds = new Set<Id>();
+		for (CW_Schedule__c s :schedules) scheduleIds.add(s.Id);
+		
+		// update schedules
+		for (CW_Campaign__c c :camps) c.Stage__c = 'stage 2';
+		update camps;
+		
+		// assert schedules have been updated
+		List<CW_Schedule__c> updatedSchedules = [select Id, Campaign_Stage__c from CW_Schedule__c where Id in:scheduleIds];
+		
+		for (CW_Schedule__c s :updatedSchedules) system.assertEquals('stage 2', s.Campaign_Stage__c);
+	}
+	
+	static testMethod void testRecurrenceType()
+	{
+		CW_Campaign__c camp = TestDataUtil.createCampaign();	// insert with no recurrence to avoid triggers
+    	insert camp;
+    	camp.Recurrence_Type__c = ConstantUtil.CAMPAIGN_RECURRENCE_TYPE_RECURRING;
+    	
+    	CW_Campaign__c campNew = camp.clone(true,true);
+    	campNew.Recurrence_Type__c = ConstantUtil.CAMPAIGN_RECURRENCE_TYPE_SINGLE;
+    	
+    	CW_Schedule__c sch = TestDataUtil.createSchedule(camp);
+    	insert sch;
+    	
+    	Test.startTest();
+    	
+    	CW_CampaignTriggerHandler cth = new CW_CampaignTriggerHandler(true);
+    	cth.cascadeDeleteSchedulesOnUpdate(new List<CW_Campaign__c>{campNew}, new Map<Id,CW_Campaign__c>(new List<CW_Campaign__c>{camp}));
+    	
+    	Test.stopTest();
+    	
+    	// assert that new schdule was created and old one deleted
+    	system.assertEquals(1,[select count() from CW_Schedule__c where id != :sch.id]);
+    	system.assertEquals(0,[select count() from CW_Schedule__c where id = :sch.id]);
+	}
+	
+	static testMethod void testUpdateCampaignStageWhenBriefIsRejectedOrRecalled()
```

Comment:
Style issue: test methods should be annotated with @IsTest for better readibility